app/src/main/java/com/github/bjolidon/bootcamp/database/UserCardsRTDB.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.bjolidon.bootcamp.database
+
+import com.github.bjolidon.bootcamp.model.MagicCard
+import com.github.bjolidon.bootcamp.model.MagicLayout
+import com.github.bjolidon.bootcamp.model.MagicSet
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+
+/**
+ * This class is used to manage the user's collection of cards.
+ */
+
+class UserCardsRTDB {
+    private val db = Firebase.database.reference
+    private val user = Firebase.auth.currentUser
+    private val userCardCollection = db.child(user!!.uid)
+
+    /**
+     * Adds a card to the user's collection.
+     */
+    fun addCardToCollection(card: MagicCard) : String {
+        val cardUID = card.set.code + card.number
+        userCardCollection.child(cardUID).child("name").setValue(card.name)
+        userCardCollection.child(cardUID).child("text").setValue(card.text)
+        userCardCollection.child(cardUID).child("layout").setValue(card.layout.toString())
+        userCardCollection.child(cardUID).child("convertedManaCost").setValue(card.convertedManaCost)
+        userCardCollection.child(cardUID).child("manaCost").setValue(card.manaCost)
+        userCardCollection.child(cardUID).child("setCode").setValue(card.set.code)
+        userCardCollection.child(cardUID).child("setName").setValue(card.set.name)
+        userCardCollection.child(cardUID).child("number").setValue(card.number)
+        userCardCollection.child(cardUID).child("imageUrl").setValue(card.imageUrl)
+
+        return "Card $cardUID successfully added to collection"
+
+    }
+
+    /**
+     * Removes a card from the user's collection.
+     */
+    fun removeCardFromCollection(card: MagicCard) : String {
+        val cardUID = card.set.code + card.number
+        userCardCollection.child(cardUID).removeValue()
+        return "Card $cardUID successfully removed from collection"
+    }
+
+    /**
+     * Transforms the data obtained from the database to a MagicCard object.
+     */
+    fun transformData(data: DataSnapshot): MagicCard{

```

Comment:
Could use a json library. You can send objects directly to your DB (and they are automatically parsed).